streamlit-app/backend/app.py

```diff
@@ -0,0 +1,286 @@
+import os
+import pprint as pp
+from collections import OrderedDict, defaultdict
+
+import diff_viewer
+import pandas as pd
+import streamlit as st
+from datasets import load_from_disk
+
+DATASET_DIR_PATH_BEFORE_CLEAN_SELECT = os.getenv("DATASET_DIR_PATH_BEFORE_CLEAN_SELECT") #  "/home/lucile/data"
+OPERATION_TYPES = [
+    "Applied filter",
+    "Applied deduplication function",
+    "Applied map function",
+]
+MAX_LEN_DS_CHECKS = os.getenv("MAX_LEN_DS_CHECKS")
+
+
+def get_ds(ds_path):
+    ds = load_from_disk(ds_path)
+    return ds
+
+
+def next_idx(idx: int):
+    idx += 1
+    return idx % len(st.session_state["ds"])
```

Comment:
Personally, I thought it was great that it was rotating, because without it, if we get to example 999 you're trapped, for example :smile:

Reply:
In fact, it would be nice to be able to choose the index you want to go to directly (which is possible with this app because you don't use streaming). But I think it can be done in another feature if it's useful.

Reply:
Then let's keep it :)